tests/mmm-fulfill-shared-escrow.ts

```diff
@@ -0,0 +1,416 @@
+        allowlistAuxAccount: SystemProgram.programId,
+        sellState,
+        tokenOwnerTokenRecord: getTokenRecordPDA(
+          poolData.extraNft.mintAddress,
+          poolData.extraNft.tokenAddress,
+        ).key,
+        poolTokenRecord: getTokenRecordPDA(
+          poolData.extraNft.mintAddress,
+          poolData.poolAtaExtraNft,
+        ).key,
+        poolOwnerTokenRecord: getTokenRecordPDA(
+          poolData.extraNft.mintAddress,
+          ownerExtraNftAtaAddress,
+        ).key,
+        authorizationRules: defaultRules,
+        ...DEFAULT_ACCOUNTS,
+      })
+      .preInstructions([
+        ComputeBudgetProgram.setComputeUnitLimit({ units: MIP1_COMPUTE_UNITS }),
+      ])
+      .remainingAccounts([
+        {
+          pubkey: M2_PROGRAM,
+          isWritable: false,
+          isSigner: false,
+        },
+        {
+          pubkey: buyerSharedEscrow,
+          isWritable: true,
+          isSigner: false,
+        },
+        {
+          pubkey: poolData.nftCreator.publicKey,
+          isSigner: false,
+          isWritable: true,
+        },
+      ])
+      .transaction();
+
+    const blockhashData = await connection.getLatestBlockhash();
+    tx.feePayer = seller.publicKey;
+    tx.recentBlockhash = blockhashData.blockhash;
+    tx.partialSign(cosigner, seller);
+    await sendAndAssertTx(connection, tx, blockhashData, false);
+
+    const expectedReferralFees =
+      expectedBuyPrices.makerFeePaid.toNumber() +
+      expectedBuyPrices.takerFeePaid.toNumber();
+    const [
+      sellerBalance,
+      paymentEscrowAccount,
+      paymentEscrowBalance,
+      creatorBalance,
+      referralBalance,
+      afterBuyerSharedEscrowBalance,
+    ] = await Promise.all([
+      connection.getBalance(seller.publicKey),
+      connection.getAccountInfo(poolData.poolPaymentEscrow),
+      connection.getBalance(poolData.poolPaymentEscrow),
+      connection.getBalance(poolData.nftCreator.publicKey),
+      connection.getBalance(poolData.referral.publicKey),
+      connection.getBalance(buyerSharedEscrow),
+    ]);
+
+    assert.equal(
+      sellerBalance,
+      initSellerBalance +
+        expectedBuyPrices.sellerReceives.toNumber() -
+        SIGNATURE_FEE_LAMPORTS * 2,
+    );
+    assert.equal(
+      initBuyerSharedEscrowBalance - afterBuyerSharedEscrowBalance,
+      2.2 * LAMPORTS_PER_SOL + expectedBuyPrices.makerFeePaid.toNumber(),
+    );
+    assert.equal(paymentEscrowBalance, 0);
+    assert.isNull(paymentEscrowAccount);
+    assert.equal(
+      creatorBalance,
+      initCreatorBalance + expectedBuyPrices.royaltyPaid.toNumber(),
+    );
+    assert.equal(referralBalance, initReferralBalance + expectedReferralFees);
+
+    const poolAccountInfo = await program.account.pool.fetch(poolData.poolKey);
+    assert.equal(poolAccountInfo.sellsideAssetAmount.toNumber(), 0);
+    assert.equal(poolAccountInfo.spotPrice.toNumber(), 1.2 * LAMPORTS_PER_SOL);
+  });
```

Comment:
can we also add a test to show that the pool gets properly closed after the buy cap runs out?

Reply:
added the test